src/Avalonia.Controls/Chrome/CaptionButtons.cs

```diff
@@ -102,24 +102,40 @@ protected override void OnApplyTemplate(TemplateAppliedEventArgs e)
             
             if (e.NameScope.Find<Button>(PART_CloseButton) is { } closeButton)
             {
-                closeButton.Click += (sender, e) => OnClose();
+                closeButton.Click += (sender, e) =>
```

Comment:
Nit: this `e` shadows the existing `e`, use another name? (also applicable to all handlers below)